lib/cmdlib/backup.py

```diff
@@ -252,7 +252,7 @@ def CheckPrereq(self):
         raise errors.OpPrereqError("Unable to load destination X509 CA (%s)" %
                                    (err, ), errors.ECODE_INVAL)
 
-      (errcode, msg) = utils.VerifyX509Certificate(cert, None, None)
+      (errcode, msg) = utils.VerifyX509Certificate(cert, None, None, False)
```

Comment:
Not really, and I find what I've done an ugly hack. The problem here is that `utils.VerifyX509Certificate` checks for both, _hard_ errors and _soft_ warnings (such as _pending_ expiration and, now, deprecated algorithms).  The problem in this particular case is that the caller disables the _soft_ warnings by passing `None` in both, `warn_days` and `error_days` and then raises an exception if the function returns anything at all. I replicated this behavior, but am not completely happy with it. I think we should split the hard and soft checks in two separate functions and have only the cluster verify code use both, since everybody else seems to only care about hard failures.

Reply:
I see. I thought you're passing `False` for now because you want to support import/export to/from old clusters with old algorithms.

Fine for now, and agree that this needs rework.